### PR TITLE
Moderator view

### PR DESCRIPTION
**Pull request by RobertMatyjek**
Thorough back end testing has yet to be done.

### CODE REVIEW

File: server/src/main/java/org/springframework/laborenabler/rest/UserRestController.java [hunk 1]
```diff
@@ -250,6 +263,26 @@
         return promoteUser(username, Roles.VOUCHED);
     }
 
+
+    @RequestMapping(value = "/{username}/promote/moderator", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<User> promoteModerator(@PathVariable("username") String username, int groupID, Authentication auth) throws Exception{
```
Comment (markd315) [inline server/src/main/java/org/springframework/laborenabler/rest/UserRestController.java]: Think we've got an issue with groupID here, you can declare it as an int, sure, but we need to take it in either as a path variable, query parameter or body. I would use a path variable.

```
 "/{username}/promote/moderator/{groupID}" is the path
and add annotation
@PathVariable("groupID")
```
Also add a check for whether the user is in the group.

File: client/src/app/moderatorView/moderatorPage/moderatorPage.component.html [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      
+  </div></div>  <!-- /Container -->
+</body>
+<body id="add a user">
+  <div id="container"><div id="container-inner">
+    <h1>Add a User</h1>
+    <form action="#" method="get">
+      <fieldset>
+        <div>
+          <label for="user_new">User to Add: </label>
```
Comment (markd315) [inline client/src/app/moderatorView/moderatorPage/moderatorPage.component.html]: So does this create an entirely new account AND add it to the group? That's a good thing for the admin to have access to, but could we separate them into two different features (same panel is fine).

File: client/src/app/moderatorView/moderatorPage/moderatorPage.component.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                  let password: string = getCookie('password');
+                  console.log(getCookie('username'));
+                  console.log(getCookie('password'));
+                  let headers: Headers = new Headers();
+                 const groupId = this.route.snapshot.params['groupid'];
+
+                  headers.append("Authorization", "Basic " + btoa(username + ":" + password)); 
+                   headers.append('Content-Type', 'application/x-www-form-urlencoded');    
+
+                const url = 'http://localhost:8080/laborenabler/api/users/' + user_current.value + '/promote/moderator';
```
Comment (markd315) [inline client/src/app/moderatorView/moderatorPage/moderatorPage.component.ts]: Yeah like I said for the backend part, I think we need to have some variable scope for groupID.

The promote doesn't work for user Trusted when I inspect his stats in Postman after hitting this promote.

I don't see how it won't be anything but null unless we make it a pathVariable or something.

File: server/src/main/java/org/springframework/laborenabler/rest/GroupRestController.java [hunk 4]
```diff
@@ -8,45 +8,53 @@
...
-      
+    public ResponseEntity<Collection<Group>> getEnrolledGroups(Authentication auth){
         Collection<Group> enrolledGroups = new ArrayList<Group>();
         enrolledGroups.add(this.resourceService.findGroupById(3));
         if (enrolledGroups.isEmpty()){
             return new ResponseEntity<Collection<Group>>(HttpStatus.NOT_FOUND);
         }
         Group currentGroup = this.resourceService.findGroupById(3);
+        //TODO this is terrible. Why did we hardcode a groupID?
+        //TODO refer to resourceService.climbGroupTree(user.getMember_of()); using auth
```
Comment (markd315) [inline server/src/main/java/org/springframework/laborenabler/rest/GroupRestController.java]: See my comment here. This backend method doesn't behave like it should at all tbh, not sure when it was written.

File: client/src/app/moderatorView/moderatorPage/moderatorPage.component.html (hunk 2 above)
Comment (RobertMatyjek) [reply client/src/app/moderatorView/moderatorPage/moderatorPage.component.html]: Yeah, I haven't implemented the back end for this, or any functionality. That will go to a different ticket and branch, just wanted to do some quick visuals for planning purposes.

**Pull request merged**